jberet-core/src/main/java/org/jberet/runtime/runner/ChunkRunner.java

```diff
@@ -642,7 +642,12 @@ private void doCheckpoint(final ProcessingInfo processingInfo) throws Exception
 
                     //usually the transaction should not be rolled back upon skippable exception, but if the transaction
                     //is marked rollback only by other parties, it's no longer usable and so roll it back.
-                    if (tm.getStatus() == Status.STATUS_MARKED_ROLLBACK || tm.getStatus() == Status.STATUS_ROLLEDBACK) {
+                    //
+                    //JBERET-597 (skippable exception thrown in ItemWriter triggers new chunk within the old transaction)
+                    //The skippable exception is from ItemWriter so the current chunk will be skipped.
+                    //There is no reason to keep active transactions as they may linger around after this skipped chunk.
+                    if (tm.getStatus() == Status.STATUS_ACTIVE ||
+                            tm.getStatus() == Status.STATUS_MARKED_ROLLBACK || tm.getStatus() == Status.STATUS_ROLLEDBACK) {
```

Comment:
May introduce a breaking change to the current behavior of `l.onSkipWriteItem`

Reply:
@aandreevh  at line 634, `l.onSkipWriteItem(outputList, e);` already happened before the changed code.  Any other changes to the current behavior of skip write listener?

Reply:
If someone's `SkipWriteListener ` implementation depends on the transaction commit, now their code would break as the transaction would be rolled back. As far as I see in the `SkipWriteListener` documentation no guarantees are provided for the transaction boundaries, so would assume that is the implementator's fault of depending on an implementation specific detail but nevertheless want to note this interaction.

Reply:
@chengfang @aandreevh Is there spec that formally define the behavior on this part?

Reply:
@liweinan It's not super clear in the spec, but I think the intent is in this case, the transaction should be rolled back upon skippable exceptions in ItemWriter write.